supervised/automl.py

```diff
@@ -866,74 +461,124 @@ def load_progress(self):
+        X, y = self._build_dataframe(X, y)
+
+        self.n_features = X.shape[1]
+        self.n_classes = len(np.unique(y[~pd.isnull(y)]))
+
+        # Get attributes (__init__ params)
+        self._mode = self._get_mode()
+        self._ml_task = self._get_ml_task()
+        self._tuning_mode = self._get_tuning_mode()
+        self._results_path = self._get_results_path()
+        self._total_time_limit = self._get_total_time_limit()
+        self._model_time_limit = self._get_model_time_limit()
+        self._algorithms = self._get_algorithms()
+        self._train_ensemble = self._get_train_ensemble()
+        self._stack_models = self._get_stack_models()
+        self._eval_metric = self._get_eval_metric()
+        self._validation_strategy = self._get_validation_strategy()
+        self._verbose = self._get_verbose()
+        self._explain_level = self._get_explain_level()
+        self._golden_features = self._get_golden_features()
+        self._feature_selection = self._get_feature_selection()
+        self._start_random_models = self._get_start_random_models()
+        self._hill_climbing_steps = self._get_hill_climbing_steps()
+        self._top_models_to_improve = self._get_top_models_to_improve()
+        self._random_state = self._get_random_state()
+
+        # Must pass copies otherwise, X and y get lost
+        self._save_data(deepcopy(X), deepcopy(y))
+
+        self.verbose_print(f"AutoML current directory: {self._results_path}")
+        self.verbose_print(f"AutoML current task: {self._ml_task}")
+        self.verbose_print(f"AutoML will use algorithms : {self._algorithms}")
+        self.verbose_print(f"AutoML will optimize for metric : {self._eval_metric}")
+
+        try:
+            self.load_progress()
 
             self._start_time = time.time()
             if self._time_ctrl is not None:
                 self._start_time -= self._time_ctrl.already_spend()
 
-            if not isinstance(X_train, pd.DataFrame):
-                raise AutoMLException(
-                    "AutoML needs X_train matrix to be a Pandas DataFrame"
-                )
-
             # Automatic Exloratory Data Analysis
             if self._explain_level == 2:
-                EDA.compute(X_train, y_train, os.path.join(self._results_path, "EDA"))
-
-            self._set_ml_task(y_train)
-
-            if X_train is not None:
-                X_train = X_train.copy(deep=False)
-
-            X_train, y_train, X_validation, y_validation = self._initial_prep(
-                X_train, y_train, X_validation, y_validation
-            )
-            self._save_data(X_train, y_train, X_validation, y_validation)
-            self._set_algorithms()
-            self._set_metric()
-
-            if self._ml_task in [BINARY_CLASSIFICATION, MULTICLASS_CLASSIFICATION]:
-                self._check_imbalanced(y_train)
+                EDA.compute(
+                    deepcopy(X), deepcopy(y), os.path.join(self._results_path, "EDA")
```

Comment:
We don't need to copy the data to make EDA. There are no changes in the data in EDA going on.

Reply:
Okok! Will fix it

Reply:
If I remember, the same was for saving (somewhere in the code). Could you fix saving as well.

Reply:
Removing `deepcopy ` on saving causes EDA to crash, as `X `and `y ` get empty after the `save` method. Don't really know why `X ` and `y` get empty when passed to save method. Have to look deeper into that, but for now, `deepcopy `solves the issue.